src/utils/hooks/contactsHooks.test.ts

```diff
@@ -0,0 +1,144 @@
+
+      expect(contacts).toEqual([contact1, contact2, contact3]);
+    });
+  });
+
+  describe("useAddressExistsInContacts", () => {
+    it("returns true if the address exists in the contacts", () => {
+      const {
+        result: { current: addressExistsInContacts },
+      } = renderHook(() => useAddressExistsInContacts());
+
+      expect(addressExistsInContacts(contact2.pkh)).toEqual(true);
+    });
+
+    it("returns false if the address exists in the contacts", () => {
+      const {
+        result: { current: addressExistsInContacts },
+      } = renderHook(() => useAddressExistsInContacts());
+
+      expect(addressExistsInContacts("new_pkh")).toEqual(false);
+    });
+  });
+
+  describe("useGetContactName", () => {
+    it("returns undefined if the contact does not exist", () => {
+      const {
+        result: { current: getContactName },
+      } = renderHook(() => useGetContactName());
+
+      expect(getContactName("new_pkh")).toEqual(undefined);
+    });
+
+    it("returns the name of the contact", () => {
+      const {
+        result: { current: getContactName },
+      } = renderHook(() => useGetContactName());
+
+      expect(getContactName(contact2.pkh)).toEqual(contact2.name);
+    });
+  });
+
+  describe("useValidatePkh", () => {
+    it("returns true for a valid address", () => {
+      const {
+        result: { current: validatePkh },
+      } = renderHook(() => useValidatePkh());
+
+      expect(validatePkh(mockImplicitAddress(5).pkh)).toEqual(true);
+    });
+
+    it("fails for invalid address", () => {
+      const {
+        result: { current: validatePkh },
+      } = renderHook(() => useValidatePkh());
+
+      expect(validatePkh("invalid_pkh")).toEqual("Invalid address");
+    });
+
+    it("fails for address already registered in contacts", () => {
+      const {
+        result: { current: validatePkh },
+      } = renderHook(() => useValidatePkh());
+
+      expect(validatePkh(contact2.pkh)).toEqual("Address already registered");
+    });
+
+    it("fails for address already saved as account", () => {
+      const {
+        result: { current: validatePkh },
+      } = renderHook(() => useValidatePkh());
+
+      expect(validatePkh(contact2.pkh)).toEqual("Address already registered");
+    });
+
+    it.each([
+      { type: "ledger", pkh: mockLedgerAccount(0).address.pkh },
+      { type: "social", pkh: mockSocialAccount(1).address.pkh },
+      { type: "secret_key", pkh: mockSecretKeyAccount(2).address.pkh },
+      { type: "mnemonic", pkh: mockMnemonicAccount(3).address.pkh },
+      { type: "multisig", pkh: mockMultisigAccount(0).address.pkh },
+    ])("fails for address used for $type account", ({ pkh }) => {
+      store.dispatch(contactsActions.reset());
+      store.dispatch(accountsSlice.actions.addAccount(mockLedgerAccount(0)));
+      store.dispatch(accountsSlice.actions.addAccount(mockSocialAccount(1)));
+      store.dispatch(accountsSlice.actions.addAccount(mockSecretKeyAccount(2)));
```

Comment:
there is a neat test helper called just `addAccount` which should shorten all of these

Reply:
Good one, fixed